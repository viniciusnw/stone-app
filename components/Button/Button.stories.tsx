import React from "react";
import { Text } from "react-native";
import { Button, StorybookWrapper } from "@Components";

const ButtonDefault = ({ type }: any) => (
  <StorybookWrapper>
    <Button color={type} onPress={() => true}>
      <Text>Default Button</Text>
    </Button>
  </StorybookWrapper>
);

export const Default = {
  args: {
    type: "default",
  },
};

export const Primary = {
  args: {
    type: "primary",
  },
};

export const Black = {
  args: {
    type: "black",
  },
};

export const Gray = {
  args: {
    type: "gray",
  },
};

export default {
  title: "Button",
  argTypes: {
    type: {
      options: ["default", "primary", "black", "gray"],
      control: { type: "radio" },
    },
  },
  component: ({ type }: any) => <ButtonDefault type={type} />,
};
