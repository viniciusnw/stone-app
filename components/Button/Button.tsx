import React from "react";
import styled from "styled-components/native";
import { Text, TouchableOpacity } from "react-native";

const StyledButton = styled(TouchableOpacity)<{ color: any }>`
  background-color: ${({ theme, color }) =>
    theme.color[!color ? "default" : color]};
  height: 40px;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 10px 15px;
`;

const StyledText = styled(Text)`
  color: ${({ theme }) => theme.color.white};
  font-weight: 700;
  font-size: 14px;
`;

const Button = ({ color, children, ...props }: any) => {
  return (
    <StyledButton color={color} {...props}>
      <StyledText>{children}</StyledText>
    </StyledButton>
  );
};

export default Button;
