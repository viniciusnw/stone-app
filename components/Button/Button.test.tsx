import React from "react";
import renderer from "react-test-renderer";
import Button from "./Button";
import { darkTheme, lightTheme } from "@Theme";
import { ThemeProvider } from "styled-components/native";
import "jest-styled-components";
import { Text } from "react-native";

test("ActionBar", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={darkTheme}>
        <Button color="primary" onPress={() => true}>
          <Text>Default Button</Text>
        </Button>
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
