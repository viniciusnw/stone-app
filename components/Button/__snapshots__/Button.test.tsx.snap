// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionBar 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgb(10,245,61)",
      "display": "flex",
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 10,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "#ffffff",
        "fontSize": 14,
        "fontWeight": "700",
      }
    }
  >
    <Text>
      Default Button
    </Text>
  </Text>
</View>
`;
